### PR TITLE
Prepare androidauto-v0.12.0

### PR DESCRIPTION
**Pull request by kmadsen**
# Update

Updating the dependencies before the release. This moves the androidauto sdk to navigation 2.9. It updates the map extension to 0.3.0 which is using the maps 10.9.0-beta. It also removes search as an inherited library.

This is matching the versions that nav-androidauto sdk will use for a stable release.

Also note this known issue will be fixed by a 2.9.0 release https://github.com/mapbox/mapbox-navigation-android/pull/6392

# Changelog

#### Bug fixes and improvements
- Use `MapboxAudioGuidance` from public api. [#6336](https://github.com/mapbox/mapbox-navigation-android/pull/6336)
- Fixed an issue that caused camera animations to interfere with user gestures. [#6357](https://github.com/mapbox/mapbox-navigation-android/pull/6357)
- Marked `CarFeedbackSender`, `CarGridFeedbackScreen`, `CarLocationsOverviewCamera`, `CarNavigationCamera`, `PlacesListItemClickListener`, `PlacesListOnMapLayerUtil`, `PlacesListOnMapScreen`, `CarRoutePreviewScreen`, `CarRouteRequest`, `PlaceSearchScreen`, `MainCarScreen`, `AppAudioGuidanceUtil`, `MapboxCarApp` methods with `@UiThread` annotation. [#6267](https://github.com/mapbox/mapbox-navigation-android/pull/6267)
- Removed `MapboxNavigation` from `CarRouteRequest` constructor. [#6372](https://github.com/mapbox/mapbox-navigation-android/pull/6372)
- Removed Search SDK as an inherited library. You must include search directly in order to access the api. [#6388](https://github.com/mapbox/mapbox-navigation-android/pull/6388)

### Mapbox dependencies
This release defines minimum versions for the Mapbox dependencies.
- Mapbox Maps Android Auto Extension `v0.3.0` ([release notes](https://github.com/mapbox/mapbox-maps-android/releases/tag/extension-androidauto-v0.3.0))
- Mapbox Navigation `v2.9.0-alpha.3` ([release notes](https://github.com/mapbox/mapbox-navigation-android/releases/tag/v2.9.0-alpha.3))
- Mapbox Search `v1.0.0-beta.36` ([release notes](https://github.com/mapbox/mapbox-search-android/releases/tag/v1.0.0-beta.36))

### CODE REVIEW

File: LICENSE.md [hunk 1]
```diff
@@ -4021,6 +4021,12 @@ License: [The Apache Software License, Version 2.0](http://www.apache.org/licens
 
 ===========================================================================
 
+Mapbox Navigation uses portions of the Activity Kotlin Extensions (Kotlin extensions for 'activity' artifact).
```
Comment (Zayankovsky) [inline LICENSE.md:4024]: All of the added sections are duplicated now, why are these changes necessary?

File: libnavui-androidauto/build.gradle [hunk 2]
```diff
@@ -41,13 +41,10 @@ dependencies {
...
     // build.gradle.
-    def carNavVersion = "2.8.0-rc.1"
+    def carNavVersion = "2.9.0-alpha.3"
     def carSearchVersion = "1.0.0-beta.36"
     api("com.mapbox.search:mapbox-search-android:${carSearchVersion}")
-
-    // TODO use the carNavVersion when 2.9.0-alpha.2 is available
-    api(project(":libnavigation-android"))
-//    api("com.mapbox.navigation:android:${carNavVersion}")
+    api("com.mapbox.navigation:android:${carNavVersion}")
```
Comment (Zayankovsky) [inline libnavui-androidauto/build.gradle:47]: We used to have a project dependency with debug build type (#6316), have we abandoned this approach?

File: LICENSE.md (hunk 1 above)
Comment (kmadsen) [reply LICENSE.md:4024]: Honestly was letting the license script do the work, it was failing https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-android/20930/workflows/078c27e8-d0d5-4775-9e9b-0e5147b411ab/jobs/106520

File: libnavui-androidauto/build.gradle (hunk 2 above)
Comment (kmadsen) [reply libnavui-androidauto/build.gradle:47]: We needed that in order to support incompatible releases. When we push code that is incompatible it also blocks releases. This is a big part of the reason to ensure compatible apis, so that we can always have a release pipeline that is not blocked.

File: LICENSE.md (hunk 1 above)
Comment (Zayankovsky) [reply LICENSE.md:4024]: We should investigate, I see that kotlinx-coroutines-core is included 17 times here!

File: libnavui-androidauto/CHANGELOG.md [hunk 3]
```diff
@@ -5,11 +5,21 @@ Mapbox welcomes participation and contributions from everyone.
...
+#### Features
+#### Bug fixes and improvements
 - Use `MapboxAudioGuidance` from public api. [#6336](https://github.com/mapbox/mapbox-navigation-android/pull/6336)
 - Fixed an issue that caused camera animations to interfere with user gestures. [#6357](https://github.com/mapbox/mapbox-navigation-android/pull/6357)
 - Marked `CarFeedbackSender`, `CarGridFeedbackScreen`, `CarLocationsOverviewCamera`, `CarNavigationCamera`, `PlacesListItemClickListener`, `PlacesListOnMapLayerUtil`, `PlacesListOnMapScreen`, `CarRoutePreviewScreen`, `CarRouteRequest`, `PlaceSearchScreen`, `MainCarScreen`, `AppAudioGuidanceUtil`, `MapboxCarApp` methods with `@UiThread` annotation. [#6267](https://github.com/mapbox/mapbox-navigation-android/pull/6267)
 - Removed `MapboxNavigation` from `CarRouteRequest` constructor. [#6372](https://github.com/mapbox/mapbox-navigation-android/pull/6372)
 
+### Mapbox dependencies
+This release defines minimum versions for the Mapbox dependencies.
+- Mapbox Maps Android Auto Extension `v0.2.0` ([release notes](https://github.com/mapbox/mapbox-maps-android/releases/tag/extension-androidauto-v0.2.0))
```
Comment (Zayankovsky) [inline libnavui-androidauto/CHANGELOG.md:19]: Will you update to 0.3.0?

Comment (kmadsen) [reply libnavui-androidauto/CHANGELOG.md:19]: Yeah might as well, the release is happening now but it's still building

It will also require some api changes

Comment (kmadsen) [reply libnavui-androidauto/CHANGELOG.md]: It's updated

File: libnavui-androidauto/CHANGELOG.md [hunk 4]
```diff
@@ -5,11 +5,21 @@ Mapbox welcomes participation and contributions from everyone.
...
 - Fixed an issue that caused camera animations to interfere with user gestures. [#6357](https://github.com/mapbox/mapbox-navigation-android/pull/6357)
 - Marked `CarFeedbackSender`, `CarGridFeedbackScreen`, `CarLocationsOverviewCamera`, `CarNavigationCamera`, `PlacesListItemClickListener`, `PlacesListOnMapLayerUtil`, `PlacesListOnMapScreen`, `CarRoutePreviewScreen`, `CarRouteRequest`, `PlaceSearchScreen`, `MainCarScreen`, `AppAudioGuidanceUtil`, `MapboxCarApp` methods with `@UiThread` annotation. [#6267](https://github.com/mapbox/mapbox-navigation-android/pull/6267)
 - Removed `MapboxNavigation` from `CarRouteRequest` constructor. [#6372](https://github.com/mapbox/mapbox-navigation-android/pull/6372)
 
+### Mapbox dependencies
+This release defines minimum versions for the Mapbox dependencies.
+- Mapbox Maps Android Auto Extension `v0.3.0` ([release notes](https://github.com/mapbox/mapbox-maps-android/releases/tag/extension-androidauto-v0.3.0))
+- Mapbox Navigation `v2.9.0-alpha.3` ([release notes](https://github.com/mapbox/mapbox-navigation-android/releases/tag/v2.8.0-beta.3))
```
Comment (Zayankovsky) [inline libnavui-androidauto/CHANGELOG.md:20]: Links for Nav and Search SDKs need to be updated.

**Review by Zayankovsky — APPROVED** [review]

File: libnavui-androidauto/build.gradle (hunk 2 above)
Comment (kmadsen) [reply libnavui-androidauto/build.gradle:47]: Need to remove the `api` from the search dependency though

File: LICENSE.md (hunk 1 above)
Comment (kmadsen) [reply LICENSE.md:4024]: OK figured out the cause https://github.com/mapbox/mapbox-navigation-android/pull/6336/files#diff-4673a3aba01813b595de187a7a6e9e63a3491d55821606fecd9f13a10c188a1dL4066-L4083

it's in the difference between `api(project(":libnavigation-android"))` and `api("com.mapbox.navigation:android:${carNavVersion}")`

When including an `api(project` the licenses are merged. When including a library with the specific version, all the dependencies are duplicated.

Considering androidauto is a libarary in isolation, considering the idea that we generate a separate license.

But for now I think this duplication is fine.

Comment (Zayankovsky) [reply LICENSE.md:4024]: Looks like it affects all of the modules in the project, not just Android Auto. Which explains why kotlinx-coroutines-core is included 17 times.
